### ISSUE TITLE
When clicking Autocomplete's Select All really fast, everything crashes.

### ISSUE THREAD

**Issue by mhwaage** [equinor/design-system#3426]
**Describe the bug**

When clicking Autocomplete's Select All really fast, everything crashes.

**Steps to reproduce the bug**

1. Have a convoluted enough enironment that there is lag when selecting options in Autocomplete
2. Set up a controlled Autocomplete component
3.  Click Select All twice, faster than the component can update between
4. Crash

**Expected behavior**
No crash

**Specifications**

- Version: 0.36.1
- Browser: Edge/firefox
- OS: Windows/Mac

**Additional context**
![image](https://github.com/equinor/design-system/assets/57612883/91eff74f-6a28-42b5-896d-5de34f46ff86)
Crash is reported as happening in:
![image](https://github.com/equinor/design-system/assets/57612883/74fd6871-2731-4874-a45d-ea2d84dc7a72)

**Comment by mhwaage** [comment]
I'll try to experiment with resolving locally, since it has proven difficult to set up an MRE.

**Comment by oddvernes** [comment]
I found the [commit](https://github.com/equinor/design-system/pull/1987/commits/d41466737a678fa10c5292f1cd87a9fabf60ff27) where this useEffect "if (isControlled)" was added from [this](https://github.com/equinor/design-system/pull/1987) pull request. Unfortunately it doesn't us anything about why it was added, just "`Also found some bugs related to controlled that was fixed 🐛`". Im going to test more if removing this doesn't have any side effects. Unfortunately havent been able to reproduce the crash yet
